module/platform/darwin/SensorFilter/src/SensorFilter.cpp

```diff
@@ -399,28 +410,70 @@ namespace module::platform::darwin {
+                        // Use our load sensor to work out which foot is down
+                        feet_down = load_sensor.updateFeet(*sensors);
 
-                    if (this->config.debug) {
-                        emit(graph("Sensor/Foot Down/Load/Left", feet_down[BodySide::LEFT]));
-                        emit(graph("Sensor/Foot Down/Load/Right", feet_down[BodySide::RIGHT]));
+                        if (this->config.debug) {
+                            emit(graph("Sensor/Foot Down/Load/Left", feet_down[BodySide::LEFT]));
+                            emit(graph("Sensor/Foot Down/Load/Right", feet_down[BodySide::RIGHT]));
+                        }
+                    }
+                    else {
+                        // TODO These metrics should probably be different
+                        if (input.left.fsr1 < -config.footDown.certaintyFSRThreshold
+                            && input.left.fsr2 < -config.footDown.certaintyFSRThreshold
+                            && input.left.fsr3 < -config.footDown.certaintyFSRThreshold
+                            && input.left.fsr4 < -config.footDown.certaintyFSRThreshold
+                            && input.right.fsr1 > config.footDown.certaintyFSRThreshold
+                            && input.right.fsr2 > config.footDown.certaintyFSRThreshold
+                            && input.right.fsr3 > config.footDown.certaintyFSRThreshold
+                            && input.right.fsr4 > config.footDown.certaintyFSRThreshold) {
+                            feet_down[BodySide::RIGHT] = true;
+                            feet_down[BodySide::LEFT]  = false;
+                        }
+                        // Right foot is above left foot in left foot space by more than the certainty threshold
+                        else if (input.left.fsr1 > config.footDown.certaintyFSRThreshold
+                                 && input.left.fsr2 > config.footDown.certaintyFSRThreshold
+                                 && input.left.fsr3 > config.footDown.certaintyFSRThreshold
+                                 && input.left.fsr4 > config.footDown.certaintyFSRThreshold
+                                 && input.right.fsr1 < -config.footDown.certaintyFSRThreshold
+                                 && input.right.fsr2 < -config.footDown.certaintyFSRThreshold
+                                 && input.right.fsr3 < -config.footDown.certaintyFSRThreshold
+                                 && input.right.fsr4 < -config.footDown.certaintyFSRThreshold) {
+                            feet_down[BodySide::RIGHT] = false;
+                            feet_down[BodySide::LEFT]  = true;
+                        }
+                        // Right foot and left foot are roughly the same height in left foot space
+                        else {
+                            feet_down[BodySide::RIGHT] = true;
+                            feet_down[BodySide::LEFT]  = true;
+                        }
```

Comment:
We used to calculate centre of pressure from the FSRs. Should be able to do something with centre of pressure being inside the foot's polygon or something?

Reply:
https://github.com/NUbots/NUbots/blob/0063ef1a00874bcb790bc27c9d518c47c8b6c091/module/platform/darwin/SensorFilter/src/SensorFilter.cpp#L584-L588

https://github.com/NUbots/NUbots/blob/0063ef1a00874bcb790bc27c9d518c47c8b6c091/shared/utility/motion/ForwardKinematics.h#L556-L574

Reply:
Didn't know we knew the polygon, but thats a better way to do it. Will still need a threshold?

Reply:
Didn't @MattAmos do something about feet polygons?

Reply:
Is that in the codebase though? Or matlab?

Reply:
Not sure